wcfsetup/install/files/js/WoltLabSuite/Core/Ui/Reaction/Handler.js

```diff
@@ -0,0 +1,375 @@
+				}
+				
+				UiAlignment.set(this._getPopover(), element, {
+					pointer: true,
+					horizontal: (this._options.isButtonGroupNavigation) ? 'left' :'center',
+					vertical: 'top'
+				});
+				
+				this._popoverCurrentObjectId = objectId;
+				
+				if (this._options.isButtonGroupNavigation) {
+					// find nav element
+					var nav = element.closest('nav');
+					nav.style.opacity = "1";
+				}
+				
+				this._getPopover().classList.remove('forceHide');
+				this._getPopover().classList.add('active');
+			},
+			
+			/**
+			 * Returns the react popover element. 
+			 * 
+			 * @returns {Element}
+			 */
+			_getPopover: function() {
+				if (this._popover == null) {
+					this._popover = elCreate('div');
+					this._popover.className = 'reactionPopover forceHide';
+					
+					var _popoverContent = elCreate('div');
+					_popoverContent.className = 'reactionPopoverContent';
+					
+					var popoverContentHTML = elCreate('ul');
+					
+					var sortedReactionTypes = this._getSortedReactionTypes();
+					
+					for (var key in sortedReactionTypes) {
+						if (!sortedReactionTypes.hasOwnProperty(key)) continue;
+						
+						var reactionType = sortedReactionTypes[key];
+						
+						var reactionTypeItem = elCreate('li');
+						reactionTypeItem.className = 'reactionTypeButton jsTooltip';
+						elData(reactionTypeItem, 'reaction-type-id', reactionType.reactionTypeID);
+						elData(reactionTypeItem, 'title', reactionType.title);
+						reactionTypeItem.title = reactionType.title;
+						
+						var reactionTypeItemSpan = elCreate('span');
+						reactionTypeItemSpan.classList = 'reactionTypeButtonTitle';
+						reactionTypeItemSpan.innerHTML = reactionType.title;
+						
+						reactionTypeItem.innerHTML = reactionType.renderedIcon;
+						
+						reactionTypeItem.appendChild(reactionTypeItemSpan);
+						
+						reactionTypeItem.addEventListener(WCF_CLICK_EVENT, this._react.bind(this, reactionType.reactionTypeID));
+						
+						popoverContentHTML.appendChild(reactionTypeItem);
+					}
+					
+					_popoverContent.appendChild(popoverContentHTML);
+					this._popover.appendChild(_popoverContent);
+					
+					var pointer = elCreate('span');
+					pointer.className = 'elementPointer';
+					pointer.appendChild(elCreate('span'));
+					this._popover.appendChild(pointer);
+					
+					document.body.appendChild(this._popover);
+					
+					DomChangeListener.trigger();
+				}
+				
+				return this._popover; 
+			},
+			
+			/**
+			 * Sort the reaction types by the showOrder field. 
+			 * 
+			 * @returns     {Array}         the reaction types sorted by showOrder
+			 */
+			_getSortedReactionTypes: function() {
+				var sortedReactionTypes = [];
+				
+				for (var key in REACTION_TYPES) {
+					if (!REACTION_TYPES.hasOwnProperty(key)) continue;
+					var reactionType = REACTION_TYPES[key];
+					
+					sortedReactionTypes[reactionType.showOrder] = reactionType;
```

Comment:
This might create an array with holes. I suggest converting the Object into an Array first, then calling `.sort` on that Array.